page-objects/smartTable.ts

```diff
@@ -0,0 +1,69 @@
+import { Locator, Page } from '@playwright/test';
+
+interface RowData {
+  id: string;
+  firstName: string;
+  lastName: string;
+  username: string;
+  email: string;
+  age: string;
+}
+export class SmartTablePage {
+  readonly page: Page;
+  readonly tableRows: Locator;
+
+  constructor(page: Page) {
+    this.page = page;
+
+    // Chọn tất cả hàng trong bảng (tbody > tr)
+    this.tableRows = page.locator('table tbody tr');
+  }
+
+  /**
+   * Trả về số lượng dòng (tr) trong bảng
+   */
+  async getRowCount(): Promise<number> {
+    return await this.tableRows.count();
+  }
+
+  /**
+   * Lấy nội dung text từng dòng, trả về dưới dạng mảng string[]
+   */
+  async getRowTexts(): Promise<string[]> {
+    // Đợi ít nhất 1 dòng hiển thị
+    await this.tableRows.first().waitFor({ state: 'visible', timeout: 5000 });
```

Comment:
Maybe we should handle this one a bit, by
if (the 1st line appears) {
   handle the logic
} handle error, adding the console.log here to cross check